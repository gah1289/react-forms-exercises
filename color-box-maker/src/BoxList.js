import React, { useState } from 'react';
import NewBoxForm from './NewBoxForm';
import Box from './Box';
import { v4 as uuid } from 'uuid';

const BoxList = () => {
	const [
		boxes,
		setBoxes
	] = useState('');

	const createBox = (newBox) => {
		setBoxes((boxes) => [
			...boxes,
			{ newBox }
		]);
	};

	function showBoxes() {
		if (boxes.length > 0) {
			return (
				<div>
					{' '}
					{boxes.map((b) => (
						<Box
							key={uuid()}
							height={b.newBox.height}
							width={b.newBox.width}
							backgroundColor={b.newBox.backgroundColor}
						/>
					))}
				</div>
			);
		}
	}

	return (
		<div>
			<h1> Color Box Maker</h1>
			<NewBoxForm createBox={createBox} />
			{showBoxes()}
		</div>
	);
};

export default BoxList;
