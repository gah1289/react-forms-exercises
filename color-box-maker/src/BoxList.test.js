import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';

function addBox(boxList, height = '200', width = '200', color = 'blue') {
	const heightInput = boxList.getByLabelText('Box Height (px)');
	const widthInput = boxList.getByLabelText('Box Width (px)');
	const backgroundInput = boxList.getByLabelText('Background Color');
	fireEvent.change(backgroundInput, { target: { value: color } });
	fireEvent.change(widthInput, { target: { value: width } });
	fireEvent.change(heightInput, { target: { value: height } });
	const button = boxList.getByText('Add Box');
	fireEvent.click(button);
}

it('renders without crashing', function() {
	render(<BoxList />);
});

it('matches snapshot', function() {
	const { asFragment } = render(<BoxList />);
	expect(asFragment()).toMatchSnapshot();
});

it('can add a new item', function() {
	// box does not exists yet
	expect(screen.queryByText('x')).not.toBeInTheDocument();
});

it('creates a box', function() {
	// box exists
	const boxList = render(<BoxList />);
	addBox(boxList);
	expect(screen.getByText('x')).toBeInTheDocument();
});
