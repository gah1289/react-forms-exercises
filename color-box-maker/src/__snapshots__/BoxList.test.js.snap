// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
       Color Box Maker
    </h1>
    <form>
      <label
        for="height"
      >
        Box Height (px)
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Box Width (px)
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Background Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
